hal/avr32/sonar_i2cxl.c

```diff
@@ -96,13 +98,41 @@ void sonar_i2cxl_get_last_measure(sonar_i2cxl_t* sonar_i2cxl)
 	
 	if ( distance_m > sonar_i2cxl->data.min_distance && distance_m < sonar_i2cxl->data.max_distance )
 	{
+		dt = ((float)time_us - sonar_i2cxl->data.last_update)/1000000.0f;
+
+		if (sonar_i2cxl->data.healthy)
+		{
+			velocity = (distance_m - sonar_i2cxl->data.current_distance) / dt;
+			if (abs(velocity)>20.0f)
+			{
+				velocity = 0.0f;
+			}
+			
+			if (sonar_i2cxl->data.healthy_vel)
+			{
+				sonar_i2cxl->data.current_velocity = (1.0f-LPF_SONAR_VARIO)*sonar_i2cxl->data.current_velocity + LPF_SONAR_VARIO*velocity;
+			}
+			else
```

Comment:
@ndousse 
I don't think that else condition is required, since if the sensor is healthy the velocity will be healthy as well, isn't it ?